quesma/schema/registry.go

```diff
@@ -126,46 +183,57 @@ func (s *schemaRegistry) populateSchemaFromDynamicConfiguration(indexName string
 }
 
 func (s *schemaRegistry) AllSchemas() map[IndexName]Schema {
-	if schemas, err := s.loadSchemas(); err != nil {
-		logger.Error().Msgf("error loading schemas: %v", err)
-		return make(map[IndexName]Schema)
-	} else {
-		return schemas
-	}
+	s.Lock()
+	defer s.Unlock()
+
+	return s.loadOrGetSchemas()
 }
 
 func (s *schemaRegistry) FindSchema(name IndexName) (Schema, bool) {
-	if schemas, err := s.loadSchemas(); err != nil {
-		logger.Error().Msgf("error loading schemas: %v", err)
-		return Schema{}, false
-	} else {
-		schema, found := schemas[name]
-		return schema, found
-	}
+	s.Lock()
+	defer s.Unlock()
+
+	schemas := s.loadOrGetSchemas()
```

Comment:
`loadOrGetSchemas()` might return nil. I think we should check against this condition